### DISCUSSION TITLE
Guidance on upgrade from old Arduino CNC

### DISCUSSION THREAD

**Discussion by petere101** [phil-barrett/grblHAL-teensy-4.x#57] [Q&A] [answered]
Hi All,
First post!
I have built a BEAST of a  CNC mill. It probably weighs over 200kg (440lb)! Mainly 100x50x6 RHS aluminium with a 10mm bottom plate. It's been sitting idle for a few years and was initially set up using an Arduino Nano running GRBL. It's time for the T4.1 upgrade. I'm confused about which binary to select.

There are 4 steppers with a double being on the Y axis. The controllers for the Y axis are wired together. They stay in synch pretty well, but require the occasional 'manual tweak'. All the limit switches are proximity sensors (Hall) and work accurately each time for homing etc. My system is basically the same as this setup image 
![image](https://user-images.githubusercontent.com/102562079/160509730-4b597a9e-03bd-45e4-b80a-c00f2cf0233a.png).

And here is some quick and dirty pics
![20220329_114146](https://user-images.githubusercontent.com/102562079/160511282-f898c4f3-d55f-4d4c-9e21-ca4b2938a1ca.jpg)
![20220329_114043](https://user-images.githubusercontent.com/102562079/160511320-ce5c5b18-ca1f-4299-8d33-13d22f7660e1.jpg)

So, what is the best pre-built binary for my brand new yet to be delivered board, assuming shipping to Australia resumes?
Thanks for your input

**Comment by phil-barrett** [comment, marked as answer]
That's a sweet looking piece of steel there. 

There are several binaries that support ganged Y motors:
grblHAL_T41 3X gangY qei.hex
grblHAL_T41 3X gangY AutoSq qei.hex
grblHAL_T41 3X gangY AutoSq qei SDCard Keypad.hex

The first one does not have auto squaring enabled, the other two do.  The last one has SDCard and Keypad plug-ins enable. I recommend you set up to do auto squaring as it allows you to keep your ganged axis square by homing.  Some [details here](https://www.grbl.org/single-post/it-s-hip-to-be-auto-square).

If you want something else, you could install the grblHAL source and build yourself or contact me.

It looks like you already have opto isolators.  While they are somewhat redundant with some features of the T41U5XBB, if they handle 12 or 24V for your sensors, it would be simplest to keep them.

Shipping to Australia via US Postal Service is still not happening.  I would be willing to use UPS but there are some downsides to that.  In particular, there are additional fees - VAT and/or customs duties and probably a brokerage fee.  I have tried to find out what the charges are beforehand and get lost in the UPS system.  I tried to get an agent to walk me through it but they were useless.  I've never used UPS to Aus but the several times I did to European countries the brokerage fees  and VAT was around 40 USD.  Adding in the UPS shipping cost, it came to about 80 USD for the package - kind of an insult on a 50 USD purchase. For that reason I don't make the option available in general.  But will do it on a case by case basis. Contact me [via tindie](https://www.tindie.com/products/philba/grblhal-teensy-41-bob-special-offer/) if you want move forward.

**Reply by petere101** [reply]
Thanks for that.
I'm expecting a UPS delivery from Digikey on Thursday which will include the T4.1. Order $AUD60 ($USD45) and it was free shipping I know there are some quite excited CNC'ers downunder, and I'm wondering how we can skirt around the ridiculous freight charges. Let me think
